File: 9.cs
using System;

bool restartGame = true;
int cpuScore = 0, userScore = 0;
string username;

do
{
    bool flag = false, flag2 = false, flag3 = false, flag5 = false;
    string difficulty, limit, guess, playAgain;
    char charLimit, charPlayAgain;
    int varDifficulty, varLimit = 0, attempts = 0, varGuess;

    Console.WriteLine("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
    Console.WriteLine(" _______________________________________________________________\n|\t\t\t\t\t\t\t\t|");
    Console.WriteLine("|\t\t\tNUMBER GUESSING GAME\t\t\t|");
    Console.WriteLine("|_______________________________________________________________|\n\n");

    Console.Write("ENTER YOUR USERNAME: ");
    username = Console.ReadLine();
    username = username.Trim();
    
    do
    {
        Console.WriteLine("\nSELECT DIFFICULTY:\n\n-EASY(1)\n-MEDIUM(2)\n-HARD(3)\n-EXIT(4)\n");
        difficulty = Console.ReadLine();
        difficulty = difficulty.Trim();
        flag = Int32.TryParse(difficulty, out varDifficulty);
        if (!flag || varDifficulty < 1 || varDifficulty > 4)
        {
            Console.WriteLine("\n\nINVALID INPUT.\n");
            flag = true;
        }
        else if (varDifficulty == 4) System.Environment.Exit(0);
        else
        {
            flag = false;
        }
        
    } while (flag == true);
    
    do
    {
        Console.WriteLine("\n\n\nCHOOSE LIMIT:\n\n***10(A)***\n***100(B)***\n***1000(C)***\n***EXIT(D)***\n");
        limit = Console.ReadLine();
        limit = limit.Trim();
        charLimit = limit[0];
        switch (charLimit)
        {
            case 'a': 
                flag2 = false;
                charLimit = Char.ToUpper(charLimit);
                break;
            case 'A':
                flag2 = false;
                break;
            case 'b':
                flag2 = false;
                charLimit = Char.ToUpper(charLimit);
                break;
            case 'B':
                flag2 = false;
                break;
            case 'c':
                flag2 = false;
                charLimit = Char.ToUpper(charLimit);
                break;
            case 'C':
                flag2 = false;
                break;
            case 'd':
                flag2 = false;
                Console.WriteLine("\nDO YOU WANT TO EXIT?\nYES(Y)\nNO(N)");
                string confirmation;
                char charConfirmation;
                bool flag4 = false;
                confirmation = Console.ReadLine();
                confirmation = confirmation.Trim();
                charConfirmation = confirmation[0];
                do
                {
                    if (charConfirmation == 121 || charConfirmation == 89) System.Environment.Exit(0);
                    else if (charConfirmation == 110 || charConfirmation == 78)
                    {
                        flag2 = true;
                        flag4 = false;  
                    }
                    else
                    {
                        flag4 = true;
                        Console.WriteLine("\n\nINVALID INPUT\nDO YOU WANT TO EXIT?\nYES(Y)\nNO(N)");
                        confirmation = Console.ReadLine();
                        confirmation = confirmation.Trim();
                        charConfirmation = confirmation[0];
                    }
                } while (flag4 == true);
                break;
            case 'D':
                flag2 = false;
                System.Environment.Exit(0);
                break;
            default:
                Console.Write("\nINVALID INPUT.");
                flag2 = true;
                break;
        }
    } while (flag2 == true);

    if (charLimit == 65) varLimit = 10;
    else if (charLimit == 66) varLimit = 100;
    else if (charLimit == 67) varLimit = 1000;

    Console.Write("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nDIFFICULTY: ");
    switch (varDifficulty)
    {
        case 1:
            Console.Write("EASY");
            break;
        case 2:
            Console.Write("MEDIUM");
            break;
        default:
            Console.Write("HARD");
            break;
    }
    Console.Write("\t\t\t\t\t\tLIMIT: " + varLimit);
    Console.Write("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nPRESS ENTER TO CONTINUE OR ANY KEY TO EXIT: ");
    if (Console.ReadKey().Key != ConsoleKey.Enter)
    {
        System.Environment.Exit(0);
    }
    
    Console.Write("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nSCORE: CPU " + cpuScore + " - " + userScore + " " + username);
    Console.Write("\t\t\t\t\t\tNUMBER OF ATTEMPTS: ");
    switch (varDifficulty)
    {
        case 1:
            if (varLimit == 10) attempts = 4;//4
            else if (varLimit == 100) attempts = 10;//10
            else if (varLimit == 1000) attempts = 15;//15
            break;
        case 2:
            if (varLimit == 10) attempts = 3;//3
            else if (varLimit == 100) attempts = 6;//6
            else if (varLimit == 1000) attempts = 10;//10
            break;
        default:
            if (varLimit == 10) attempts = 2;//2
            else if (varLimit == 100) attempts = 4;//4
            else if (varLimit == 1000) attempts = 8;//8
            break;
    }
    Console.Write(attempts);

    Random randomNum = new Random();
    int varRandomNum = randomNum.Next(varLimit + 1);

    do
    {
        Console.Write("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nMAKE A GUESS: ");
        guess = Console.ReadLine();
        guess = guess.Trim();
        flag3 = Int32.TryParse(guess, out varGuess);
        if (!flag3 || varGuess != varRandomNum)
        {
            attempts -= 1;
            Console.Write("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nSCORE: CPU " + cpuScore + " - " + userScore + " " + username);
            Console.Write("\t\t\t\t\t\tNUMBER OF ATTEMPTS: " + attempts);
            if (varGuess < varRandomNum) Console.Write("\n\n\nYOU ARE LOOKING FOR A BIGGER NUMBER");
            else if (varGuess > varRandomNum) Console.Write("\n\n\nYOU ARE LOOKING FOR A SMALLER NUMBER");
        }
        if (attempts == 0) break;
    } while (varGuess != varRandomNum);

    if (attempts == 0)
    {
        cpuScore += 1;
        Console.WriteLine("\n\n\nGAME OVER! YOU LOST, " + username + "!" + "\nTHE NUMBER WAS " + varRandomNum + "\n\n\n\n");
        Console.WriteLine("SCCORE: CPU " + cpuScore + " - " + userScore + " " + username + "\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
    }
    else if (attempts != 0)
    {
        userScore += 1;
        Console.WriteLine("\n\n\nYOU WON, " + username + "! " + "THE NUMBER WAS " + varRandomNum + "\n\n\n\n\n");
        Console.WriteLine("SCORE: CPU " + cpuScore + " - " + userScore + " " + username + "\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
    }

    do
    {
        Console.Write("DO YOU WANT TO PLAY AGAIN? YES(Y) NO(N): ");
        playAgain = Console.ReadLine();
        charPlayAgain = playAgain[0];
        switch (charPlayAgain)
        {
            case 'Y':
                flag5 = false;
                break;
            case 'y':
                flag5 = false;
                break;
            case 'N':
                restartGame = false;
                flag5 = false;
                break;
            case 'n':
                restartGame = false;
                flag5 = false;
                break;
            default:
                flag5 = true;
                break;
        }
    } while (flag5 == true);

} while (restartGame == true);

Console.Write("\nTHANKS FOR PLAYING, " + username + "!");
Console.WriteLine("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");

return 0;

File: 9.py
import random, sys

restartGame = True
username = ""
cpuScore = 0
user_score = 0  

while restartGame == True:
    print("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n")
    print("_______________________________________________________________\n|\t\t\t\t\t\t\t\t|")
    print("|\t\t\tNUMBER GUESSING GAME\t\t\t|")
    print("|_______________________________________________________________|\n\n")

    username = input("ENTER YOUR USERNAME: ")
    username = username.strip()

    while True:
        difficulty = input("SELECT DIFFICULTY:\n\n-EASY(1)\n-MEDIUM(2)\n-HARD(3)\n-EXIT(4)\n\n")
        difficulty = difficulty.strip()
        if difficulty.isdigit():
            var_difficulty = int(difficulty)
            if var_difficulty < 1 or var_difficulty > 4:
                print("\n\nINVALID INPUT.\n")
            elif var_difficulty == 4:
                sys,exit()
            else:
                break
        else:
            print("\n\nINVALID INPUT.\n")
        
    while True:
        limit = input("\n\n\nCHOOSE LIMIT:\n\n***10(A)***\n***100(B)***\n***1000(C)***\n***EXIT(D)***\n\n")
        limit = limit.strip()
        if not limit.isalpha():
            print("\nINVALID INPUT.")
        else:
            limit = limit.lower()
        if limit == 'a':
            var_limit = 10
            break
        elif limit == 'b':
            var_limit = 100
            break
        elif limit == 'c':
            var_limit = 1000
            break
        elif limit == 'd':
            while True:
                confirmation = input("\nDO YOU WANT TO EXIT?\nYES(Y)\nNO(N)\n")
                confirmation = confirmation.strip()
                if not confirmation.isalpha():
                    print("\n\nINVALID INPUT")
                else:
                    confirmation = confirmation.lower()                
                if confirmation == 'n':
                    break
                elif confirmation == 'y': 
                    sys.exit()
                else: 
                    print("\n\nINVALID INPUT")
        else:
            print("\nINVALID INPUT.")

    print("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nDIFFICULTY: ", end='')
    if var_difficulty == 1:
        print("EASY", end='') 
    elif var_difficulty == 2:
        print("MEDIUM", end='')
    else:
        print("HARD", end='')
    
    print("\t\t\t\t\t\tLIMIT: {0}".format(var_limit))
    key = input("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nPRESS ENTER TO CONTINUE OR ANY KEY TO EXIT: ")
    if key != '':
        sys.exit()  
    
    print("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nSCORE: CPU {0} - {1} {2}".format(cpuScore, user_score, username), end="")
    print("\t\t\t\t\t\tNUMBER OF ATTEMPTS: ", end="")
    if var_difficulty == 1:
        if var_limit == 10: 
            attempts = 4
        elif var_limit == 100:
            attempts = 10
        else:
            attempts = 15
    elif var_difficulty == 2:
        if var_limit == 10:
            attempts = 3
        elif var_limit == 100:
            attempts = 6
        else:
            attempts == 10
    else:
        if var_limit == 10:
            attempts = 2
        elif var_limit == 100:
            attempts = 4
        else:
            attempts = 8
    print(attempts)

    varRandomNum = random.randint(1, var_limit)

    while True:
        guess = input("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nMAKE A GUESS: ")
        guess = guess.strip()
        if not guess.isdigit():
            attempts -= 1
            print("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nSCORE: CPU {0} - {1} {2}".format(cpuScore, user_score, username), end="")
            print("\t\t\t\t\t\tNUMBER OF ATTEMPTS: {0}".format(attempts))
        else:
            varGuess = int(guess)
            if varGuess != varRandomNum:
                attempts -= 1
                print("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nSCORE: CPU {0} - {1} {2}".format(cpuScore, user_score, username), end="")
                print("\t\t\t\t\t\tNUMBER OF ATTEMPTS: {0}".format(attempts))
                if varGuess < varRandomNum:
                    print("\n\n\nYOU ARE LOOKING FOR A BIGGER NUMBER")
                else:
                    print("\n\n\nYOU ARE LOOKING FOR A SMALLER NUMBER")
        if attempts == 0 or varGuess == varRandomNum:
            break

    if attempts == 0:
        cpuScore += 1
        print("\n\n\nGAME OVER! YOU LOST, {0}!".format(username))
        print("THE NUMBER WAS {0}\n\n\n\n".format(varRandomNum))
        print("SCORE: CPU {0} - {1} {2}\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n".format(cpuScore, user_score, username))
    else:
        user_score += 1
        print("\n\n\nYOU WON, {0}! THE NUMBER WAS {1}\n\n\n\n\n".format(username, varGuess))

    
    while True:
        playAgain = input("DO YOU WANT TO PLAY AGAIN? YES(Y) NO(N): ")
        playAgain = playAgain.strip().upper()
        if playAgain == 'Y':
            break
        if playAgain == 'N':
            restartGame = False
            break

print("Thanks for playing, " + username + '!')
print("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n")



    
